From 46e9d93e020deeb9247e9fe396f50f9485168b66 Mon Sep 17 00:00:00 2001
From: Nikolay Makarov <makarov-na@yandex.ru>
Date: Sun, 5 Apr 2026 14:20:17 +0300
Subject: [PATCH] custom keyboard

---
 ru | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/ru b/ru
index ff821f9..fd0a7ad 100644
--- a/ru
+++ b/ru
@@ -1256,6 +1256,39 @@ xkb_symbols "winkeys-p" {
     key <FK12> { [ F12,         bar ] };
 };

+// RU Winkeys-PD layout based on Russian (Programmer)
+// Changes:
+//     -Classic top row digits
+//     -Shift+F1 &
+//     -Shift+F2 :
+//     -Shift+F7 ~
+// Nikolay Makarov <makarov-na@yandex.ru>
+partial alphanumeric_keys
+xkb_symbols "winkeys-pd" {
+    include "ru(winkeys)"
+    name[Group1]= "Russian (Programmer classic digits)";
+
+    key <AE01> { [ 1, exclam        ] };
+    key <AE02> { [ 2, at            ] };
+    key <AE03> { [ 3, numbersign    ] };
+    key <AE04> { [ 4, dollar        ] };
+    key <AE05> { [ 5, percent       ] };
+    key <AE06> { [ 6, asciicircum   ] };
+    key <AE07> { [ 7, ampersand     ] };
+    key <AE08> { [ 8 ,asterisk      ] };
+    key <AE09> { [ 9 ,parenleft     ] };
+    key <AE10> { [ 0, parenright    ] };
+    key <FK01> { [ F1,          ampersand ] };
+    key <FK02> { [ F2,          colon ] };
+    key <FK07> { [ F7,          asciitilde ] };
+    key <FK06> { [ F6,          numerosign ] };
+    key <FK08> { [ F8,          quotedbl ] };
+    key <FK09> { [ F9,          apostrophe ] };
+    key <FK11> { [ F11,         question ] };
+    key <FK12> { [ F12,         bar ] };
+};
+
+
 // Russian keyboard layout with typography symbols.
 // This is a modified Russian standard keyboard, with the third level containing
 // typographic symbols commonly used in Russian language and punctuation symbols
--
2.54.0
