--- ru_orig	2026-04-05 14:42:13.119396401 +0300
+++ ru	2026-07-11 14:45:19.890740340 +0300
@@ -1256,6 +1256,34 @@
     key <FK12> { [ F12,         bar ] };
 };
 
+// RU Winkeys-PD layout based on Russian (Programmer)
+// Changes:
+//     -Classic top row digits
+    //     -Shift+F1 &
+//     -Shift+F2 :
+//     -Shift+F7 ~
+// Nikolay Makarov <makarov-na@yandex.ru>
+partial alphanumeric_keys
+xkb_symbols "winkeys-pd" {
+    include "ru(winkeys-p)"
+    name[Group1]= "Russian (Programmer classic digits)";
+
+    key <AE01> { [ 1, exclam        ] };
+    key <AE02> { [ 2, at            ] };
+    key <AE03> { [ 3, numbersign    ] };
+    key <AE04> { [ 4, dollar        ] };
+    key <AE05> { [ 5, percent       ] };
+    key <AE06> { [ 6, asciicircum   ] };
+    key <AE07> { [ 7, ampersand     ] };
+    key <AE08> { [ 8 ,asterisk      ] };
+    key <AE09> { [ 9 ,parenleft     ] };
+    key <AE10> { [ 0, parenright    ] };
+    key <FK01> { [ F1,          ampersand ] };
+    key <FK02> { [ F2,          colon ] };
+    key <FK07> { [ F7,          asciitilde ] };
+};
+
+
 // Russian keyboard layout with typography symbols.
 // This is a modified Russian standard keyboard, with the third level containing
 // typographic symbols commonly used in Russian language and punctuation symbols
